### PR TITLE
Add minimal cairo 1 template

### LINKED ISSUE #1946: Add minimal cairo 1 template
For now, `protostar init` creates a following structure:
```
my_project
├── hello_starknet
│   ├── cairo_project.toml
│   └── src
│       ├── business_logic
│       │   └── utils.cairo
│       ├── business_logic.cairo
│       ├── contracts
│       │   └── hello_starknet.cairo
│       ├── contracts.cairo
│       └── lib.cairo
├── protostar.toml
└── tests
    ├── test_hello_starknet.cairo
    └── test_utils.cairo
```
All these files allow and "hint" user to create a project with multiple contracts that have mutual logic. This is good but I think we should create a minimalistic template as well.

I think we should leave the current template as is and add a flag `minimal`, so it would be invoked like this: `protostar init --minimal`. In this case, the file structure of a created project should look roughly like this:
```
my_project
├── hello_starknet
│   ├── cairo_project.toml
│   └── lib.cairo
├── protostar.toml
└── tests
    └── test_hello_starknet.cairo
```

This would be good for a quick setup for simple "foo-bar" contracts.

### PR DESCRIPTION
**Pull request by karol-bisztyga**
### Checklist
- [x] Relevant issue is linked
- [x] Docs updated/issue for docs created
- [x] Added relevant tests

### CODE REVIEW

File: website/docs/cairo-1/03-project-initialization.md [hunk 1]
```diff
@@ -164,3 +164,25 @@ fn test_returns_two() {
...
+my_project
+├── hello_starknet
+│   ├── cairo_project.toml
+│   └── lib.cairo
+├── protostar.toml
+└── tests
+    └── test_hello_starknet.cairo
+```
+
+This is way simpler than the standard structure but doesn't make it easy to develop a multi-contract project.
```
Comment (piotmag769) [inline website/docs/cairo-1/03-project-initialization.md:188]:
```suggestion
```
I would remove this comment, check the suggestion above

**Review by piotmag769 — CHANGES_REQUESTED** [review]
Please rebase to master and add `Scarb.toml` where it is needed + remove `cairo_project.toml`.
Remember that Scarb expects `src/lib.cairo` file and that we no longer support `linked-libraries` in `protostar.toml`.

File: website/docs/cairo-1/03-project-initialization.md [hunk 2]
```diff
@@ -164,3 +164,25 @@ fn test_returns_two() {
 Make sure that the path::to:the::module is correct for your package structure.
 
 For more details on of how to test contracts, see [this page](./05-testing/README.md).
+
+### Minimal project template
+
+If you don't plan to develop a complex, multi-contract project and you just want to quickly setup the easiest possible project, you can use the `--minimal` flag like this:
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply website/docs/cairo-1/03-project-initialization.md:170]: Yes but the structure "hints" you if you want to go that (multi-contract) path, that's what I meant here.

[the remark this answers is not in the document]
Comment (piotmag769) [reply website/docs/cairo-1/03-project-initialization.md:170]: I would argue it does not, the multi-contract path is described in docs and it is quite different from the default template
https://docs.swmansion.com/protostar/docs/cairo-1/project-initialization#using-multiple-contracts-in-project

File: tests/integration/commands/init_command_test/project_name_minimal/Scarb.toml
Comment (piotmag769) [inline tests/integration/commands/init_command_test/project_name_minimal/Scarb.toml:1]: I don't think including the files in `tests/integration/commands/init_command_test` are necessary, it is a leftover from previous tests. Already removed the folder in other draft PR

File: website/docs/cairo-1/03-project-initialization.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply website/docs/cairo-1/03-project-initialization.md]: [suggestions](https://github.com/software-mansion/protostar/pull/1958/commits/fa947065060198d3d96e32ea7df9e44055f1fb48)

File: website/docs/cairo-1/03-project-initialization.md (hunk 1 above)
Comment (karol-bisztyga) [reply website/docs/cairo-1/03-project-initialization.md]: [suggestions](https://github.com/software-mansion/protostar/pull/1958/commits/fa947065060198d3d96e32ea7df9e44055f1fb48)

File: website/docs/cairo-1/03-project-initialization.md [hunk 3]
```diff
@@ -164,3 +164,25 @@ fn test_returns_two() {
...
+This will generate the following file structure:
+
+```
+my_project
+├── hello_starknet
+│   ├── cairo_project.toml
+│   └── lib.cairo
+├── protostar.toml
+└── tests
+    └── test_hello_starknet.cairo
```
[the remark this answers is not in the document]
Comment (karol-bisztyga) [reply website/docs/cairo-1/03-project-initialization.md]: [update minimal example with scarb](https://github.com/software-mansion/protostar/pull/1958/commits/f1c500f14f17854e61cc1028b1eee057521a0436)

File: protostar/commands/cairo1_commands/init_command.py [hunk 4]
```diff
@@ -37,12 +37,22 @@ def arguments(self):
...
+                type="bool",
+                description="Create a minimal project structure.",
+            ),
         ]
 
     async def run(self, args: Namespace):
-        self.init(project_name=args.name)
+        minimal = False
+        if vars(args).get("minimal"):
+            minimal = bool(args.minimal)
```
Comment (cptartur) [inline protostar/commands/cairo1_commands/init_command.py:50]: `ProtostarArgument` should accept an `default=` when constructing. There's no need to solve it like this.

File: protostar/commands/cairo1_commands/init_command.py [hunk 5]
```diff
@@ -37,12 +37,22 @@ def arguments(self):
...
+        self.init(project_name=args.name, minimal=minimal)
 
-    def init(self, project_name: Optional[str] = None):
+    def init(self, project_name: Optional[str] = None, minimal: bool = False):
```
Comment (cptartur) [inline protostar/commands/cairo1_commands/init_command.py:53]: If we are setting a default in `run`, why do we also need a default value here?

File: templates/cairo1_minimal/tests/test_hello_starknet.cairo [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    let result_before = call(contract_address, 'get_balance', @ArrayTrait::new()).unwrap();
+    assert(*result_before.at(0_u32) == 0, 'Invalid balance');
+
+    let mut invoke_calldata = ArrayTrait::new();
+    invoke_calldata.append(0);
+    let invoke_result = invoke(contract_address, 'increase_balance', @invoke_calldata);
+
+    assert(invoke_result.is_err(), 'Invoke should fail');
+}
```
Comment (cptartur) [inline templates/cairo1_minimal/tests/test_hello_starknet.cairo]: Maybe if we want this template to minimal, we should consider some simpler example tests? But I'm not sure about this.

File: tests/integration/commands/init_command_test/project_name_minimal/Scarb.toml
Comment (karol-bisztyga) [reply tests/integration/commands/init_command_test/project_name_minimal/Scarb.toml:1]: [remove redundant test data](https://github.com/software-mansion/protostar/pull/1958/commits/c0d57ecac1091daf15bfdcad57566079fbe259ed)

File: templates/cairo1_minimal/tests/test_hello_starknet.cairo (hunk 6 above)
Comment (karol-bisztyga) [reply templates/cairo1_minimal/tests/test_hello_starknet.cairo]: I think tests can be like that, it's all about the structure. People are going to delete the code anyway if they want to write their own. But I don't mind that much either.

**Review by piotmag769 — APPROVED** [review]

File: protostar/commands/cairo1_commands/init_command.py (hunk 4 above)
Comment (karol-bisztyga) [reply protostar/commands/cairo1_commands/init_command.py]: [feedback](https://github.com/software-mansion/protostar/pull/1958/commits/095daf8b6e677d4336582679a059432726939c90)

File: protostar/commands/cairo1_commands/init_command.py (hunk 5 above)
Comment (karol-bisztyga) [reply protostar/commands/cairo1_commands/init_command.py]: [feedback](https://github.com/software-mansion/protostar/pull/1958/commits/095daf8b6e677d4336582679a059432726939c90)

**Review by cptartur — APPROVED** [review]

**Pull request merged**